tck/features/CreateAcceptance.feature

```diff
@@ -0,0 +1,607 @@
+    Given any graph
+    When executing query:
+      """
+      CREATE (:A:B:C:D)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes  | 1 |
+      | +labels | 4 |
+
+  Scenario: Combine MATCH and CREATE
+    Given any graph
+    And having executed:
+      """
+      CREATE (), ()
+      """
+    When executing query:
+      """
+      MATCH ()
+      CREATE ()
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes  | 2 |
+
+  Scenario: Combine MATCH, WITH and CREATE
+    Given any graph
+    And having executed:
+      """
+      CREATE (), ()
+      """
+    When executing query:
+      """
+      MATCH ()
+      CREATE ()
+      WITH *
+      MATCH ()
+      CREATE ()
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes  | 10 |
+
+  Scenario: Newly-created nodes not visible to preceding MATCH
+    Given any graph
+    And having executed:
+      """
+      CREATE ()
+      """
+    When executing query:
+      """
+      MATCH ()
+      CREATE ()
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes  | 1 |
+
+  Scenario: Create a single node with properties
+    Given any graph
+    When executing query:
+      """
+      CREATE (n {prop: 'foo'})
+      RETURN n.prop AS p
+      """
+    Then the result should be:
+      | p     |
+      | 'foo' |
+    And the side effects should be:
+      | +nodes      | 1 |
+      | +properties | 1 |
+
+  Scenario: Creating a node with null properties should not return those properties
+    Given any graph
+    When executing query:
+      """
+      CREATE (n {id: 12, property: NULL})
+      RETURN n.id AS id
+      """
+    Then the result should be:
+      | id |
+      | 12 |
+    And the side effects should be:
+      | +nodes      | 1 |
+      | +properties | 1 |
+
+  Scenario: Creating a relationship with null properties should not return those properties
+    Given any graph
+    When executing query:
+      """
+      CREATE ()-[r:X {id: 12, property: NULL}]->()
+      RETURN r.id
+      """
+    Then the result should be:
+      | r.id |
+      | 12   |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+      | +properties    | 1 |
+
+  Scenario: Create a simple pattern
+    Given any graph
+    When executing query:
+      """
+      CREATE ()-[:R]->()
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+
+  Scenario: Create a self loop
+    Given any graph
+    When executing query:
+      """
+      CREATE (root:R)-[:LINK]->(root)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 1 |
+      | +relationships | 1 |
+      | +labels        | 1 |
+
+  Scenario: Create a self loop using MATCH
+    Given any graph
```

Comment:
And the last case, when there's a node property existence constraint on label `:R`, this won't work.